pkg/apis/core/validation/validation.go

```diff
@@ -5195,17 +5196,31 @@ func ValidatePodTemplateSpec(spec *core.PodTemplateSpec, fldPath *field.Path, op
 	return allErrs
 }
 
-func ValidateReadOnlyPersistentDisks(volumes []core.Volume, fldPath *field.Path) field.ErrorList {
+// ValidateReadOnlyPersistentDisks validates pd reads and writes
+func ValidateReadOnlyPersistentDisks(featureGate featuregate.FeatureGate, volumes, oldVolumes []core.Volume, fldPath *field.Path, isNew bool) field.ErrorList {
 	allErrs := field.ErrorList{}
+
+	if !featureGate.Enabled(features.SkipReadOnlyValidationGCE) {
```

Comment:
Don't you want the opposite of this?  If the gate is enabled, we skip validation.

We should add a unit test for this function covering:

* gate enabled with read-only disk, nil old => OK
* gate enabled with read-write disk, nil old => OK
* gate disabled with read-only disk, nil old => OK
* gate disabled with read-write disk, nil old => error
* gate disabled with new read-only and old read-only => OK
* gate disabled with new read-only and old read-write => OK
* gate disabled with new read-write and old read-only => error
* gate disabled with new read-write and old read-write => OK

Reply:
yes sir

Reply:
```
if featureGate.Enabled(features.SkipReadOnlyValidationGCE) {
		return field.ErrorList{}
}
```
I found that panic appears in unit test after adding feature gates

Reply:
well, we'll have to figure that out :)

Reply:
I have passed all the requirements tests